Model/Api/ShippingMethods.php

```diff
@@ -266,22 +283,26 @@ public function getShippingMethods($cart, $shipping_address)
 
             $this->hookHelper->verifyWebhook();
 
-            $incrementId = $cart['display_id'];
-
-            // Get quote from increment id
-            $quote   = $this->quoteFactory->create()->load($incrementId, 'reserved_order_id');
-            $quoteId = $quote->getId();
+            // Load quote from entity id
+            $quoteId = $cart['order_reference'];
+            $quote = $this->quoteFactory->create()->load($quoteId);
 
-            if (empty($quoteId)) {
+            if (!$quote || !$quote->getId()) {
                 throw new LocalizedException(
-                    __('Invalid display_id :%1.', $incrementId)
+                    __('Unknown quote id: %1.', $quoteId)
                 );
             }
 
-            $this->checkoutSession->replaceQuote($quote);
-
             $this->checkCartItems($cart, $quote);
 
+            if ($customerId = $quote->getCustomerId()) {
+                $this->customerSession->setCustomer(
+                    $this->customerFactory->create()->load($customerId)
```

Comment:
@YevhenShBolt 
Using QuoteRepository instead of QuoteFacctory now.
CustomerRepository instead of CustomerFactory an this place throws an error. Leaving that part the way it was.